src_common/eth_plugin_interface.h

```diff
@@ -139,19 +132,22 @@ typedef struct ethPluginProvideParameter_s {
 // Finalize
 
 typedef struct ethPluginFinalize_s {
-    ethPluginSharedRW_t *pluginSharedRW;
-    ethPluginSharedRO_t *pluginSharedRO;
-    uint8_t *pluginContext; // PLUGIN_CONTEXT_SIZE
+    // READ ONLY //
+    const txContent_t *txContent;
 
-    uint8_t *tokenLookup1;  // set by the plugin if a token should be looked up
-    uint8_t *tokenLookup2;
+    // READ WRITE //
+    uint8_t *pluginContext; // PLUGIN_CONTEXT_SIZE
 
-    const uint8_t *amount;   // set an uint256 pointer if uiType is UI_AMOUNT_ADDRESS
-    const uint8_t *address;  // set to the destination address if uiType is UI_AMOUNT_ADDRESS. Set
-                             // to the user's address if uiType is UI_TYPE_GENERIC
+    // WRITE ONLY //
+    const uint8_t *tokenLookup1;  // set by the plugin if a token should be looked up
+    const uint8_t *tokenLookup2;
 
     eth_ui_type_t uiType;
-    uint8_t numScreens;  // ignored if uiType is UI_AMOUNT_ADDRESS
+    const uint8_t *amount;  // set an uint256 pointer if uiType is UI_AMOUNT_ADDRESS
+    const uint8_t *address; // set to the destination address if uiType is UI_AMOUNT_ADDRESS.
+                            // set to the user's address if uiType is UI_TYPE_GENERIC
+    uint8_t numScreens;     // set the number of screens to display if uiType is UI_TYPE_GENERIC
```

Comment:
Could these fields managed inside a `union` (they seem to be exclusive)?

Reply:
Done